src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -102,8 +111,14 @@ def self.getModules(env)
       # NOTE: If in the future we want to collect more
       #       Module settings, this should be more programatic
       #       rather than getting these settings one by one
-      module_data[:name] = mod.forge_name
-      module_data[:version] = mod.version
+      if mod.forge_name
+        module_data[:name] = mod.forge_name
+        module_data[:version] = mod.version
```

Comment:
If someone were to, probably accidentally I guess, have a "metadata.json" file in a module but leave the `version` field out of it, would this end up being assigned / returned as `nil` - and potentially fail schema validation somewhere in the Clojure code?  If so, maybe it would be better here to have the logic to assign a default value for the field based on a `nil` value for the field in the Ruby object be done separately for `name` vs. `version`.  WDYT?